From 1965c81c57de467415730b2069af90f6a3de2f07 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 12 May 2021 17:15:46 +0200
Subject: [PATCH 3/3] SSDK: platform: use of_mdio_find_bus() to get MDIO bus

Kernel has a generic of_mdio_find_bus() which can get the appropriate
MDIO bus based on the DT node.
So, drop the getting MDIO from platform data, which no longer works
in 5.4 and later and use of_mdio_find_bus().

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 src/init/ssdk_plat.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/init/ssdk_plat.c b/src/init/ssdk_plat.c
index 093b8189..f5ea7604 100755
--- a/src/init/ssdk_plat.c
+++ b/src/init/ssdk_plat.c
@@ -547,7 +547,6 @@ static int miibus_get(a_uint32_t dev_id)
 	struct device_node *mdio_node = NULL;
 	struct device_node *switch_node = NULL;
 	struct platform_device *mdio_plat = NULL;
-	struct ipq40xx_mdio_data *mdio_data = NULL;
 	struct qca_phy_priv *priv;
 	hsl_reg_mode reg_mode = HSL_REG_LOCAL_BUS;
 	priv = qca_phy_priv_global[dev_id];
@@ -580,12 +579,7 @@ static int miibus_get(a_uint32_t dev_id)
 
 	if(reg_mode == HSL_REG_LOCAL_BUS)
 	{
-		mdio_data = dev_get_drvdata(&mdio_plat->dev);
-		if (!mdio_data) {
-			SSDK_ERROR("cannot get mdio_data reference from device data\n");
-			return 1;
-		}
-		priv->miibus = mdio_data->mii_bus;
+		priv->miibus = of_mdio_find_bus(mdio_node);
 	}
 	else
 		priv->miibus = dev_get_drvdata(&mdio_plat->dev);
-- 
2.31.1

